lib/adhearsion/reporter/email_notifier.rb

```diff
@@ -33,6 +34,14 @@ def exception_text(exception)
         exception.backtrace.join("\n") +
         "\n\n"
       end
+
+      def environment
+        Adhearsion.config.platform.environment.to_s.upcase
+      end
+
+      def hostname
+        Socket.gethostname
+      end
```

Comment:
Shame on me for not having done this in my own code, but can you `require 'socket'` to ensure that we have Socket available?  Part of the stdlib, so no need for any additions to Gemfile, fortunately.